x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/repository_list/repository_details/repository_details.tsx

```diff
@@ -318,6 +314,81 @@ export const RepositoryDetails: React.FunctionComponent<Props> = ({
     </Fragment>
   );
 
+  const renderCleanup = () => (
+    <>
+      <EuiTitle size="s">
+        <h3>
+          <FormattedMessage
+            id="xpack.snapshotRestore.repositoryDetails.cleanupTitle"
+            defaultMessage="Repository cleanup"
+          />
+        </h3>
+      </EuiTitle>
+      <EuiSpacer size="s" />
+      <EuiText size="s">
+        <p>
+          <FormattedMessage
+            id="xpack.snapshotRestore.repositoryDetails.cleanupRepositoryMessage"
+            defaultMessage="You can clean up a repository to delete any unreferenced data from a snapshot. This
+              may provide storage space savings. Note: If you regularly delete snapshots, this
+              functionality will likely not be as beneficial and should be used less frequently."
+          />
+        </p>
+      </EuiText>
+      {cleanup ? (
+        <>
+          <EuiSpacer size="s" />
+          {cleanup.cleaned ? (
+            <div>
+              <EuiTitle size="xs">
+                <h4>
+                  <FormattedMessage
+                    id="xpack.snapshotRestore.repositoryDetails.cleanupDetailsTitle"
+                    defaultMessage="Details"
+                  />
+                </h4>
+              </EuiTitle>
+              <EuiCodeBlock language="json" inline={false} data-test-subj="cleanupCodeBlock">
+                {JSON.stringify(cleanup.response, null, 2)}
+              </EuiCodeBlock>
+            </div>
+          ) : (
+            <EuiCallOut
+              color="danger"
+              iconType="alert"
+              title={i18n.translate('xpack.snapshotRestore.repositoryDetails.cleanupErrorTitle', {
+                defaultMessage: 'Sorry, there was an error cleaning the repository.',
+              })}
+            >
+              <p>
+                {JSON.stringify(
```

Comment:
I think you only need to call `JSON.stringify()` for `cleanup.error`. The default `i18n.translate()` should already be a string.

Reply:
Fixed.